core/factory.py

```diff
@@ -124,15 +124,35 @@ def is_member(self, uid: int, whitelist: List[int]) -> bool:
             # whitelist is a whitelist, we don't lock whitelisted users
             return True
         return False
-
-    def notify_about_membership(self, message) -> None:
+    
+    def is_blogger(self, message: Message) -> bool:
+        """
+        Checks if the user is writing as a channel
+        """
+        user = message.from_user
+        if user.first_name == 'Channel' and user.is_bot:
+            return True
+        return False
+    
+    def notify_about_membership(self, message, is_blogger: bool = False) -> None:
```

Comment:
Why are we modifying the existing method? That method has its own purpose. We should make a new one, something like "notify_about_violation".